src/service/viewer-impl.js

```diff
@@ -147,11 +147,11 @@ export class Viewer {
     this.params_ = {};
 
     /** @private {?function()} */
-    this.whenVisibleResolve_ = null;
+    this.whenFirstVisibleResolve_ = null;
 
     /** @private @const {!Promise} */
-    this.whenVisiblePromise_ = new Promise(resolve => {
-      this.whenVisibleResolve_ = resolve;
+    this.whenFirstVisiblePromise_ = new Promise(resolve => {
+      this.whenFirstVisibleResolve_ = resolve;
```

Comment:
This constructor calls `onVisibilityChange_` at the end which does it. But I agree, at the very least, this should be documented.